lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -757,6 +749,55 @@ class StableHLOToTTIRConcatOpConversionPattern
   }
 };
 
+template <typename SrcOp, typename DestOp,
+          typename Adaptor = typename SrcOp::Adaptor>
+class StableHLOToTTIROpLogicalOpConversionPattern
+    : public OpConversionPattern<SrcOp> {
+
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    LogicalResult err = checkBasicLegality(srcOp, adaptor, rewriter);
+    if (not err.succeeded()) {
+      return err;
+    }
+
+    auto outputType = mlir::cast<RankedTensorType>(
+        this->getTypeConverter()->convertType(srcOp.getResult().getType()));
+    tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
+        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+    rewriter.replaceOpWithNewOp<DestOp>(
+        srcOp,
+        TypeRange(
+            this->getTypeConverter()->convertType(outputTensor.getType())),
+        adaptor.getOperands(), ValueRange(outputTensor),
+        rewriter.getArrayAttr(
+            SmallVector<Attribute>(adaptor.getOperands().size() + 1,
+                                   rewriter.getAttr<OperandConstraintAttr>(
+                                       OperandConstraint::AnyDeviceTile))));
+    return success();
+  }
+
+private:
+  LogicalResult checkBasicLegality(SrcOp srcOp, Adaptor adaptor,
+                                   ConversionPatternRewriter &rewriter) const {
+    if (mlir::cast<RankedTensorType>(srcOp->getOperand(0).getType())
+                .getElementTypeBitWidth() > 1 &&
+        mlir::cast<RankedTensorType>(srcOp->getOperand(1).getType())
+                .getElementTypeBitWidth() > 1) {
+      llvm::errs()
+          << "error: TTIR does not support bitwise logical operation.\n";
```

Comment:
Can you support the bitwise operations by any combination of existing ttnn operations? If not, can you raise an issue for ttnn to add support for it?

Reply:
There is an open issue for [bitwise ops](https://github.com/tenstorrent/tt-metal/issues/13582). I'll escalate the issue if we have any blocker for bitwise ops; currently, we don't have any.